### ISSUE TITLE
The question of pre-training accuracy

### ISSUE THREAD

**Issue by bunnyveil** [LMD0311/PointMamba#16]
I first ran the classification task of ModelNet40 training-from-scratch in pointmamba on four 3080Ti, and selected the your pretrain.pth file to run the train from pre-trained classification of ModelNet40. All these are implemented in accordance with the parameters and steps described in the paper, but the final classification accuracy is only 93.0713%, while the best classification accuracy mentioned in the paper is 93.6%.We ran it many times(pic 1 with voting and pic 2,3,4 without voting) I have tested other classification and segmentation tasks, and the results obtained in accordance with the parameters in the paper have decreased by about 1-4 percentage points compared with those in the paper. I wonder if this is a problem or if the model parameters in the article are not updated in real time?
<img width="636" alt="1" src="https://github.com/LMD0311/PointMamba/assets/147686458/df2cc27f-0a77-43c7-8a5f-386a9276a12f">
<img width="672" alt="2" src="https://github.com/LMD0311/PointMamba/assets/147686458/c2cae982-4315-496f-bf9b-a13a2f3ca919">
<img width="802" alt="3" src="https://github.com/LMD0311/PointMamba/assets/147686458/45b52140-7009-4472-b3f7-f27f26341d35">
<img width="722" alt="4" src="https://github.com/LMD0311/PointMamba/assets/147686458/4c53e1cd-c27f-43c9-b65c-a33c7ef78d0f">

**Comment by LMD0311** [comment]
Thank you for your interest in our work!

The code currently open-sourced from the repository aligns with the paper's current version. Our models are all trained and tested using a single RTX 4090, which may differ across hardware platforms. Also please refer to [this issue](https://github.com/CVMI-Lab/PAConv/issues/9#issuecomment-873371422) and the [explanation here](https://github.com/ma-xu/pointMLP-pytorch?tab=readme-ov-file#model-zoo).

Feel free to evaluate our open-sourced weight for ModelNet40.

**Comment by formerlya** [comment]
May I ask if you can directly test with the weights you provide? The accuracy of the test in pointmamba is only 7%~8%, and the weight files provided by them are also used in point-MAE, with about 93.5%
<img width="909" alt="1716887100665" src="https://github.com/LMD0311/PointMamba/assets/88373367/16bb0c50-2c21-459d-a267-e38d20d0d010">

**Comment by formerlya** [comment]
并且当我训练的时候
<img width="255" alt="1716887279564" src="https://github.com/LMD0311/PointMamba/assets/88373367/fd3b3874-4720-4d06-aa5d-f2df23eaceaa">
会nan
同样的数据，在mae中可以正常训练
请问可以直接使用代码中的超参数吗，需要针对性调整？

**Comment by LMD0311** [comment]
> May I ask if you can directly test with the weights you provide? The accuracy of the test in pointmamba is only 7%~8%, and the weight files provided by them are also used in point-MAE, with about 93.5% <img alt="1716887100665" width="909" src="https://private-user-images.githubusercontent.com/88373367/334341629-16bb0c50-2c21-459d-a267-e38d20d0d010.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTY4OTc3NjgsIm5iZiI6MTcxNjg5NzQ2OCwicGF0aCI6Ii84ODM3MzM2Ny8zMzQzNDE2MjktMTZiYjBjNTAtMmMyMS00NTlkLWEyNjctZTM4ZDIwZDBkMDEwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTI4VDExNTc0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE3NDBkOTlkOTJmYzAzY2MxMzY4ZTI1ZGJmNzQ1Mjk1MTY5NTE0MTg3YTk1NzU2MTZlOGRkMzQ2YmVhM2E5OWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.7ADAOVCxtYuj_dHEDd5Mckt_LdImvuWc_6DhoIe6G98">

I just re-cloned the repository, downloaded the open-sourced checkpoint, and achieved the exact same results as reported. The test command should be `CUDA_VISIBLE_DEVICES=0 python main.py --test --config cfgs/finetune_modelnet.yaml --ckpts modelnet_scratch.pth`.

**Comment by LMD0311** [comment]
> 并且当我训练的时候 <img alt="1716887279564" width="255" src="https://private-user-images.githubusercontent.com/88373367/334343294-fd3b3874-4720-4d06-aa5d-f2df23eaceaa.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTY4OTc3NjgsIm5iZiI6MTcxNjg5NzQ2OCwicGF0aCI6Ii84ODM3MzM2Ny8zMzQzNDMyOTQtZmQzYjM4NzQtNDcyMC00ZDA2LWFhNWQtZjJkZjIzZWFjZWFhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTI4VDExNTc0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYwMTY3MWYwZDVmN2UxOTBiMGVjZjE5NWE4Y2VjZmU5YTNkZWY1YjY0MGM0OWJjMjRkMzJmMjA2YTQxYmRmNWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.jo5kq4Og5epiQ-UR3kA7f0pjqNTxiK9DUsxTfmGxPJk"> 会nan 同样的数据，在mae中可以正常训练 请问可以直接使用代码中的超参数吗，需要针对性调整？

I'm sorry, we never encountered an issue with NaN.

**Comment by formerlya** [comment]
> > 并且当我训练的时候 <img alt="1716887279564" width="255" src="https://private-user-images.githubusercontent.com/88373367/334343294-fd3b3874-4720-4d06-aa5d-f2df23eaceaa.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTY4OTc3NjgsIm5iZiI6MTcxNjg5NzQ2OCwicGF0aCI6Ii84ODM3MzM2Ny8zMzQzNDMyOTQtZmQzYjM4NzQtNDcyMC00ZDA2LWFhNWQtZjJkZjIzZWFjZWFhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTI4VDExNTc0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYwMTY3MWYwZDVmN2UxOTBiMGVjZjE5NWE4Y2VjZmU5YTNkZWY1YjY0MGM0OWJjMjRkMzJmMjA2YTQxYmRmNWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.jo5kq4Og5epiQ-UR3kA7f0pjqNTxiK9DUsxTfmGxPJk"> 会nan 同样的数据，在mae中可以正常训练 请问可以直接使用代码中的超参数吗，需要针对性调整？
> 
> I'm sorry, we never encountered an issue with NaN.

Thank you very much for your answer~

**Comment by LMD0311** [comment]
I am closing this issue. Please feel free to reopen it if necessary.

**Comment by HerculesJL** [comment]
@formerlya Hello, may I ask how you solved NaN problem?

**Comment by jelly114514** [comment]
@formerlya Hello, may I ask how you solved NaN problem?

**Comment by jelly114514** [comment]
@HerculesJL Hello, may I ask how you solved NaN problem?

**Comment by JellyZh** [comment]
@HerculesJL @formerlya Hello, may I ask how you solved NaN problem?
